streams/src/main/java/org/apache/kafka/streams/kstream/internals/graph/GraphGraceSearchUtil.java

```diff
@@ -70,27 +61,4 @@ private static long findAndVerifyWindowGrace(final GraphNode graphNode, final St
         return inheritedGrace;
     }
 
-    @SuppressWarnings("rawtypes")
-    private static Long extractGracePeriod(final GraphNode node) {
```

Comment:
Main point of this PR is to get rid of this ugly thing

Reply:
Still a good progress that we replaced four nested instanceof with just one :) Just kidding.

Reply:
haha yeah it's a step in the right direction...but it's still only just one step 😛